### PR TITLE
DCA-342: Handle permission issue on Bitbucket

### PR DESCRIPTION
**Pull request by astashys**
https://bulldog.internal.atlassian.com/browse/DCA-342

### CODE REVIEW

File: app/util/data_preparation/api/bitbucket_clients.py [hunk 1]
```diff
@@ -23,6 +33,7 @@ def get_entities(self, entity_name, batch_size, filter=None, max_results=500):
             start_at = start_at + len(response.json()['values'])
             loop_count -= 1
             if loop_count == 1:
+                # TODO variable max_results is not used, before delete it we should check if there is no bug in the code
```
Comment (ometelytsia) [inline app/util/data_preparation/api/bitbucket_clients.py]: pls create a Jira ticket and add an id to this TODO

File: app/util/data_preparation/bitbucket/prepare-data.py [hunk 2]
```diff
@@ -19,15 +19,31 @@ def generate_random_string(length=20):
...
     while perf_user_count_to_create > 0:
-        user = bitbucket_api.create_user(username=f'{DEFAULT_USER_PREFIX}-{generate_random_string(5)}')
-        perf_users.append(user)
+        client.create_user(f'{DEFAULT_USER_PREFIX}-{generate_random_string(5)}')
         perf_user_count_to_create = perf_user_count_to_create - 1
-    return bitbucket_api.get_users(username=f'{DEFAULT_USER_PREFIX}', max_results=perf_user_count)
+
+    perf_users = client.get_users(f'{DEFAULT_USER_PREFIX}', perf_users_desired_number)
+    if len(perf_users) < perf_users_desired_number:
+        raise SystemExit(f'Server returned less number of users than expected')
```
Comment (ometelytsia) [inline app/util/data_preparation/bitbucket/prepare-data.py]: we could add variables into log message

File: app/util/data_preparation/api/bitbucket_clients.py (hunk 1 above)
Comment (astashys) [reply app/util/data_preparation/api/bitbucket_clients.py]: I am going to investigate it today

File: app/util/data_preparation/bitbucket/prepare-data.py (hunk 2 above)
Comment (astashys) [reply app/util/data_preparation/bitbucket/prepare-data.py]: it is a very rare case and we could have the numbers from users in csv file and concurrency in yml file

**Pull request merged**